layouts/layouts/src/main/java/com/microsoft/device/dualscreen/layouts/FoldableLayoutController.kt

```diff
@@ -0,0 +1,361 @@
+            width = WRAP_CONTENT
+            height = MATCH_PARENT
+        }
+    }
+
+    /**
+     * Updates the containers dimensions corresponding to the dual screen mode and layout orientation
+     *
+     * @param linearLayoutOrientation The [FoldableLayout] orientation
+     * @param startScreenRect The bounds for the first display area
+     * @param endScreenRect The bounds for the second display area
+     */
+    private fun updateDimensionsForDualScreen(
+        linearLayoutOrientation: Int,
+        startScreenRect: Rect,
+        endScreenRect: Rect
+    ) {
+        if (linearLayoutOrientation == VERTICAL) {
+            resetContainersDimensions()
+
+            rootView.doOnNextLayout {
+                updateDimensionsForDualLandscape()
+            }
+        } else {
+            updateDimensionsForDualPortrait(startScreenRect, endScreenRect)
+        }
+    }
+
+    /**
+     * Updates the containers dimensions corresponding to the dual screen mode in portrait orientation
+     *
+     * @param startScreenRect The bounds for the first screen
+     * @param endScreenRect The bounds for the second screen
+     */
+    private fun updateDimensionsForDualPortrait(
+        startScreenRect: Rect,
+        endScreenRect: Rect
+    ) {
+        firstContainer.updateLayoutParams<LinearLayout.LayoutParams> {
+            width = startScreenRect.width()
+            height = MATCH_PARENT
+        }
+        secondContainer.updateLayoutParams<LinearLayout.LayoutParams> {
+            width = endScreenRect.width()
+            height = MATCH_PARENT
+        }
+    }
+
+    /**
+     * Updates the containers dimensions corresponding to the dual screen mode in landscape orientation
+     */
+    private fun updateDimensionsForDualLandscape() {
+        foldingFeature?.let { foldingFeature ->
+            val location = rootView.locationOnScreen
+            val dualScreenStartContainerHeight = foldingFeature.bounds.top - location.y
+
+            firstContainer.updateLayoutParams<LinearLayout.LayoutParams> {
+                width = MATCH_PARENT
+                height = dualScreenStartContainerHeight
+            }
+            secondContainer.updateLayoutParams<LinearLayout.LayoutParams> {
+                width = MATCH_PARENT
+                height = MATCH_PARENT
+            }
+        }
+    }
+
+    /**
+     * Updates the containers dimensions to MATCH_PARENT
+     */
+    private fun resetContainersDimensions() {
+        firstContainer.updateLayoutParams<LinearLayout.LayoutParams> {
+            width = MATCH_PARENT
+            height = MATCH_PARENT
+        }
+        secondContainer.updateLayoutParams<LinearLayout.LayoutParams> {
+            width = MATCH_PARENT
+            height = MATCH_PARENT
+        }
+    }
+}
+
+/**
+ *  Clears the content, inflate the new content and add it to the view.
+ */
+private fun ViewGroup.addContent(@LayoutRes layoutResId: Int): View? {
```

Comment:
I think indentation starts getting odd here.

Reply:
This is an extension function, so it's outside of the class